### PR TITLE
Fixed network error - hanging when not connected to internet

### PR DESCRIPTION
**Pull request by Saikrishna-Achalla**
Fixes the problem in issue #1460 which gives some cryptic error message when `sky launch` is tried with no internet connection. This fix causes the program to exit with the exception shown below:

raise exceptions.NetworkError('Could not refresh the cluster. '
sky.exceptions.NetworkError: Could not refresh the cluster. Network seems down.

### CODE REVIEW

File: sky/cli.py [hunk 1]
```diff
@@ -1325,6 +1325,7 @@ def launch(
     In both cases, the commands are run under the task's workdir (if specified)
     and they undergo job queue scheduling.
     """
+    backend_utils.check_network_connection()
```
Comment (romilbhardwaj) [inline sky/cli.py:1328]: We should wrap this in a `with ux_utils.print_exception_no_traceback():` to avoid printing a long stack trace

Comment (romilbhardwaj) [inline sky/cli.py:1328]: This prints `sky.exceptions.NetworkError: Could not refresh the cluster. Network seems down.`. We're not refreshing the cluster here. We may want to change the message to something else. Ideally we want the message in `check_network_connection` to be something generic, such as `Network seems to be down`, and then higher level methods (like this) can catch this exception and re-raise with appropriate messages (e.g., here `Failed to launch - network seems to be down.`)

Comment (romilbhardwaj) [inline sky/cli.py:1328]: We also need to catch failures when running `sky exec` and also then invoking through the python API. I'm wondering if it is better to put this check lower in the stack, perhaps in `def _execute()` in execution.py?

**Review by romilbhardwaj — COMMENTED** [review]
Thanks @Saikrishna-Achalla! Left some comments.

File: sky/cli.py (hunk 1 above)
Comment (Saikrishna-Achalla) [reply sky/cli.py]: Did end up wrapping it in this but stack trace printed is the long list of exceptions still

**Pull request closed without merging**